File: apps/web/src/lib/types/correcao.ts
export interface ResumoItem {
    valor: string;
    quantidade: number;
    valor_total: number;
}

export interface HistoricoItem {
    id: number;
    data_correcao: string;
    usuario: string;
    tipo_correcao: string;
    valor_antigo?: string;
    valor_novo?: string;
    linhas_afetadas: number;
}

export interface ResumoResponse {
    formas_pagamento: ResumoItem[];
    bandeiras: ResumoItem[];
    status: ResumoItem[];
    recebiveis: ResumoItem[];
}

export interface AtualizarRequest {
    processamento_id: string;
    campo: 'forma_pagamento' | 'bandeira' | 'status' | 'lancamento';
    valores_antigos: string[];
    valor_novo: string;
    usuario?: string;
}

export interface RemoverRequest {
    processamento_id: string;
    campo: 'forma_pagamento' | 'bandeira' | 'status' | 'lancamento';
    valores: string[];
    usuario?: string;
}

export interface FiltrosBCResponse {
    formas: string[];
    bandeiras: string[];
}

export interface AplicarTaxaBCRequest {
    processamento_id: string;
    forma_pagamento: string;
    bandeira: string;
    data_ini?: string;
    data_fim?: string;
    nova_taxa: number;
    usuario?: string;
}

File: apps/api/app/schemas/correcao.py
from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel


class ResumoItem(BaseModel):
    valor: str
    quantidade: int
    valor_total: float

class HistoricoItem(BaseModel):
    id: int
    data_correcao: datetime
    usuario: str
    tipo_correcao: str
    valor_antigo: Optional[str] = None
    valor_novo: Optional[str] = None
    linhas_afetadas: int

class ResumoResponse(BaseModel):
    formas_pagamento: List[ResumoItem]
    bandeiras: List[ResumoItem]
    status: List[ResumoItem]
    recebiveis: List[ResumoItem]

class AtualizarRequest(BaseModel):
    processamento_id: str
    campo: str  # 'forma_pagamento', 'bandeira', 'status', 'lancamento'
    valores_antigos: List[str]
    valor_novo: str

class RemoverRequest(BaseModel):
    processamento_id: str
    campo: str  # 'forma_pagamento', 'bandeira', 'status', 'lancamento'
    valores: List[str]

class FiltrosBCResponse(BaseModel):
    formas: List[str]
    bandeiras: List[str]

class AplicarTaxaBCRequest(BaseModel):
    processamento_id: str
    forma_pagamento: str  # 'TODOS' ou valor específico
    bandeira: str         # 'TODOS' ou valor específico
    data_ini: Optional[str] = None
    data_fim: Optional[str] = None
    nova_taxa: float
    usuario: str = "sistema"
